### PR TITLE
Goto def code moves

### PR DESCRIPTION
**Pull request by keyboardDrummer**
Code moves in preparation for #2734 

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyCore/AST/SubstitutingCloner.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+using System.Collections.Generic;
+
+namespace Microsoft.Dafny;
+
+class SubstitutingCloner : Cloner {
```
Comment (MikaelMayer) [inline Source/DafnyCore/AST/SubstitutingCloner.cs:5]: Where is this file used? I can't find any mention of it in old code.

**Review by MikaelMayer — APPROVED** [review]
Great job at doing all these refactorings ! I can't wait to see syntax highlight again.
Just one question I could not find an answer for. If the answer is straightforward, just enable auto-merge after answering it, otherwise, I deactivated auto-merge so that you can answer adequatedly.

File: Source/DafnyCore/Resolver/Resolver.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 namespace Microsoft.Dafny {
   public partial class Resolver {
-    readonly BuiltIns builtIns;
+    public readonly BuiltIns builtIns;
```
[the remark this answers is not in the document]
Comment (keyboardDrummer) [reply Source/DafnyCore/Resolver/Resolver.cs:24]: Will follow-up with another PR after https://github.com/dafny-lang/dafny/pull/2734

File: Source/DafnyCore/Rewriters/ForallStmtRewriter.cs [hunk 3]
```diff
@@ -0,0 +1,316 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+
+namespace Microsoft.Dafny;
+
+public class ForallStmtRewriter : IRewriter {
```
[the remark this answers is not in the document]
Comment (keyboardDrummer) [reply Source/DafnyCore/Rewriters/ForallStmtRewriter.cs:7]: Added again in https://github.com/dafny-lang/dafny/pull/2734

File: Source/DafnyCore/AST/SubstitutingCloner.cs (hunk 1 above)
Comment (keyboardDrummer) [reply Source/DafnyCore/AST/SubstitutingCloner.cs:5]: Removed in https://github.com/dafny-lang/dafny/pull/2734

**Pull request merged**